pkg/gohai/cpu/cpu_darwin.go

```diff
@@ -8,49 +8,67 @@ package cpu
 import (
 	"fmt"
 
+	"github.com/DataDog/datadog-agent/pkg/gohai/utils"
 	"golang.org/x/sys/unix"
 )
 
-var cpuMapString = map[string]string{
-	"machdep.cpu.vendor":       "vendor_id",
-	"machdep.cpu.brand_string": "model_name",
+func storeSysctl[U any, V any](call func(string) (U, error), cast func(U) V, field *utils.Value[V], key string) {
```

Comment:
Did you consider having `getSysctl[T](call func(string) (T, error), key string) T` so the caller doesn't have to define a `cast` function (The caller can do the conversion without a callback)?

Reply:
What do you think about making `getSysctlString` returning `Value[string], error` instead of `string, error`?